research/minigo/minigo.py

```diff
@@ -305,38 +278,189 @@ def evaluate(trained_models_dir, black_model_name, white_model_name,
   params = model_params.MiniGoParams()
-  k = utils.round_power_of_two(board_size ** 2 / 3)
+  k = utils.round_power_of_two(FLAGS.board_size ** 2 / 3)
   params.num_filters = k  # Number of filters in the convolution layer
   params.fc_width = 2 * k  # Width of each fully connected layer
-  params.num_shared_layers = board_size  # Number of shared trunk layers
-  params.board_size = board_size  # Board size
+  params.num_shared_layers = FLAGS.board_size  # Number of shared trunk layers
+  params.board_size = FLAGS.board_size  # Board size
 
   # How many positions can fit on a graphics card. 256 for 9s, 16 or 32 for 19s.
-  if FLAGS.board_size == 9:
-    params.batch_size = 256
+  if FLAGS.batch_size is None:
+    if FLAGS.board_size == 9:
+      params.batch_size = 256
+    else:
+      params.batch_size = 32
   else:
-    params.batch_size = 32
+    params.batch_size = FLAGS.batch_size
+
   return params
 
 
+def _prepare_selfplay(
+    model_name, trained_models_dir, selfplay_dir, holdout_dir, sgf_dir, params):
+  """Set directories and load the network for selfplay.
+
+  Args:
+    model_name: The name of the model for self-play
+    trained_models_dir: Directories where the completed generations/models are.
+    selfplay_dir: Where to write the games. Set as 'base_dir/data/selfplay/'.
+    holdout_dir: Where to write the holdout data. Set as
+      'base_dir/data/holdout/'.
+    sgf_dir: Where to write the sgf (Smart Game Format) files. Set as
+      'base_dir/sgf/'.
+    params: An object of hyperparameters for the model.
+
+  Returns:
+    The directories and network model for selfplay.
+  """
+  print('Playing a game with model {}'.format(model_name))
+  # Set paths for the model with 'model_name'
+  model_path = os.path.join(trained_models_dir, model_name)
+  output_dir = os.path.join(selfplay_dir, model_name)
+  holdout_dir = os.path.join(holdout_dir, model_name)
+  # clean_sgf is to write sgf file without comments.
+  # full_sgf is to write sgf file with comments.
+  clean_sgf = os.path.join(sgf_dir, model_name, 'clean')
+  full_sgf = os.path.join(sgf_dir, model_name, 'full')
+
+  _ensure_dir_exists(output_dir)
+  _ensure_dir_exists(holdout_dir)
+  _ensure_dir_exists(clean_sgf)
+  _ensure_dir_exists(full_sgf)
+  selfplay_dirs = {
+      'output_dir': output_dir,
+      'holdout_dir': holdout_dir,
+      'clean_sgf': clean_sgf,
+      'full_sgf': full_sgf
+  }
+  # cache the network model for self-play
+  with utils.logged_timer('Loading weights from {} ... '.format(model_path)):
+    network = dualnet.DualNetRunner(model_path, params)
+  return selfplay_dirs, network
+
+
+def run_single_step_only(params, dirs):
+  """Run each individual step only.
+
+  Args:
+    params: An object of hyperparameters for the model.
+    dirs: An object of directories used in each step.
+
+  Raises:
+    ValueError: when neither of the black model and the white model is provided
+    for evaluation.
+  """
+  # bootstrap only
+  if FLAGS.bootstrap_only:
```

Comment:
I think this function is long enough and with several layer of if-else. I think we should extract the body of each top level if and put them into function calls. Eg.

if FLAGS.bootstrap_only:
  _handle_bootstrap()
  return
if FLAGS.selfplay_only:
  _handle_selfplay()
....

Reply:
Sure. Will do the refactoring.